adr/0002-workflow-integration-points.md

```diff
@@ -0,0 +1,35 @@
+# 2. Provide workflow integration points
+
+Date: 2018-04-27
+
+## Status
+
+Proposed
+
+## Context
+
+A system to handle the creation, processing and orchestration of customized workflows is a required component of Libero's
+currently proposed design.
+
+Users will each have their individual needs and requirements and therefore may decide to use their 
+workflow solution of choice. 
+
+It is therefore important to define clear integration points between a workflow system and the rest of the Libero
+architecture.
+
+## Decision
+
+The workflow solution should provide:
+
+- HTTP endpoint(s) to allow the triggering of workflows
+- A queue (AMQP 0.9.1) to allow the triggering of workflows
+- Publishing of workflow and activity state event messages via queues (AMQP 0.9.1)
+- Adhere to the [Libero event schema](https://github.com/libero/walking-skeleton/blob/master/events/schema.md)
```

Comment:
These two points are strictly necessary for the observability of what is happening inside the system. The schema could be propagated with any messaging system. We could envision that if someone wanted webhooks, for example, it would be easy to turn the event messages into HTTP requests with an adapter delivering them.